pkg/gitlab_hooks/mr_actions.go

```diff
@@ -24,16 +24,21 @@ func (w *GitlabEventWorker) processMergeRequestEvent(msg *MergeRequestEventMsg)
 		return projectName, nil
 	}
 
-	trigger := tfc_trigger.NewTFCTrigger(w.gl, w.tfc, w.runstream,
-		&tfc_trigger.TFCTriggerConfig{
-			Action:                   tfc_trigger.PlanAction,
-			Branch:                   event.ObjectAttributes.SourceBranch,
-			CommitSHA:                event.ObjectAttributes.LastCommit.ID,
-			ProjectNameWithNamespace: event.ObjectAttributes.Source.PathWithNamespace,
-			MergeRequestIID:          event.ObjectAttributes.IID,
-			TriggerSource:            tfc_trigger.MergeRequestEventTrigger,
-			VcsProvider:              "gitlab",
-		})
+	cfg, err := tfc_trigger.NewTFCTriggerConfig(&tfc_trigger.TFCTriggerOptions{
+		Action:                   tfc_trigger.PlanAction,
+		Branch:                   event.ObjectAttributes.SourceBranch,
+		CommitSHA:                event.ObjectAttributes.LastCommit.ID,
+		ProjectNameWithNamespace: event.ObjectAttributes.Source.PathWithNamespace,
+		MergeRequestIID:          event.ObjectAttributes.IID,
+		TriggerSource:            tfc_trigger.MergeRequestEventTrigger,
+		VcsProvider:              "gitlab",
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("could not create TFCTriggerConfig")
+		return projectName, err
```

Comment:
We should wrap this as a Permanent Error unless it can recovered. `utils.CreatePermanentError` since that'll put it back on the Nats queue.

Reply:
Makes 100p sense. i moved it into the NewTFCTriggerConfig function so the error will always be wrapped with the permanent error. 👍